packages/ember/tests/routing/query_params_test.js

```diff
@@ -3351,6 +3351,40 @@ if (isEnabled('ember-routing-route-configured-query-params')) {
       promiseResolve();
     });
   });
+
+  QUnit.test('undefined queryParams is treated as empty object', function() {
+    expect(1);
+
+    App.Router.map(function() {
+      this.route('foo');
+      this.route('bar');
+    });
+
+    App.FooRoute = Route.extend({
+      beforeModel() {
+        this.transitionTo('bar', { queryParams: undefined });
```

Comment:
Yes, the error only seems to occur when passing `undefined`. I still don't know what elsewhere in the system is causing something to call `transitionTo('…', { queryParams: undefined })`, but that's the cause of all the app errors.

[Here's the code which is trying to guess if we have queryParams or not](https://github.com/emberjs/ember.js/blob/67081fa/packages/ember-routing/lib/system/router.js#L359-L364):

```js
    let possibleQueryParams = args[args.length - 1];
    if (possibleQueryParams && possibleQueryParams.hasOwnProperty('queryParams')) {
      queryParams = args.pop().queryParams;
    } else {
      queryParams = {};
    }
```

So:

- `this.transitionTo('bar')` works fine without my change, because `possibleQueryParams` is falsy and so we hit the second block of the `if`

- `this.transitionTo('bar', {})` also hits the second block of the `if`, because `{}` fails the second condition, but then (for unrelated reasons) then results in "Error: More context objects were passed than there are dynamic segments for the route: bar".